controllers/kuadrant_controller.go

```diff
@@ -440,29 +440,31 @@ func (r *KuadrantReconciler) reconcileLimitador(ctx context.Context, kObj *kuadr
 
 	limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorOwnerRefsMutator)
 
-	if kObj.Spec.Limitador.Affinity != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorAffinityMutator)
-	}
-	if kObj.Spec.Limitador.Replicas != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorReplicasMutator)
-	}
-	if kObj.Spec.Limitador.Storage != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorStorageMutator)
-	}
-	if kObj.Spec.Limitador.RateLimitHeaders != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorRateLimitHeadersMutator)
-	}
-	if kObj.Spec.Limitador.Telemetry != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorTelemetryMutator)
-	}
-	if kObj.Spec.Limitador.PodDisruptionBudget != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorPodDisruptionBudgetMutator)
-	}
-	if kObj.Spec.Limitador.ResourceRequirements != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorResourceRequirementsMutator)
-	}
-	if kObj.Spec.Limitador.Verbosity != nil {
-		limitadorMutators = append(limitadorMutators, kuadranttools.LimitadorVerbosityMutator)
+	if kObj.Spec.Limitador != nil {
```

Comment:
If you feel the DRYing is worthy, maybe the reflect lib can help with iteration...

```go
if kObj.Spec.Limitador != nil {
fields := reflect.ValueOf(kObj.Spec.Limitador)
for i := 0; i < v.NumField(); i++ {
// here check if the value `v.Field(i).Interface()` is nil and call the mutator with the desired and expected
}
```